core/src/main/java/com/aem/demo/core/servlets/ServletForRssFeed.java

```diff
@@ -51,7 +29,7 @@ public class ServletForRssFeed extends SlingSafeMethodsServlet {
     protected void doGet(final SlingHttpServletRequest request,
                           final SlingHttpServletResponse response) throws IOException {
         response.setContentType("application/json");
-        rssFeedService.readFeed();
+        rssFeedService.saveRssFeedNodes(rssFeedService.readFeed());
```

Comment:
This method call looks strange. It appears that the `rssFeedService` can do two things -- read data and save data. But he, as a newborn child, cannot "decide" himself that after reading data it is required to save data! What's the sence of reading data without saving it anyhow?) 

It's like you saying to the kid, "if you put your boots on, don't sit and wait, tie them up", -- and the kid could not guess it by himself.

The solution is,
- eighter to have two different services: one for reading RSS, one for storing RSS. Then it would look and feel natural and understandable that you need to "commute" them separately;
- or else, make your service expose only one public method -- "saveData" -- and then the service _internally_ calls first "read", second, "write".